File: 051/4.js
class InsufficientBalanceException extends Error {
    constructor(message) {
      super(message);
      this.name = "InsufficientBalanceException";
    }
  }
  

  class Account {
    constructor(accountNumber, holderName, balance = 0.0) {
      this._accountNumber = accountNumber;
      this._holderName = holderName;
      this._balance = balance;
    }
  
  
    getAccountNumber() {
      return this._accountNumber;
    }
  
    setAccountNumber(accountNumber) {
      this._accountNumber = accountNumber;
    }
  
    
    getHolderName() {
      return this._holderName;
    }
  
    setHolderName(holderName) {
      this._holderName = holderName;
    }
  
   
    getBalance() {
      return this._balance;
    }
  
    setBalance(balance) {
      this._balance = balance;
    }
  
    
    deposit(amount) {
      if (amount > 0) {
        this._balance += amount;
        console.log(`ฝากเงิน ${amount} บาท สำเร็จ. ยอดเงินปัจจุบัน: ${this._balance} บาท`);
      } else {
        console.log("จำนวนเงินฝากต้องมากกว่าศูนย์");
      }
    }
  
 
    withdraw(amount) {
      if (amount > this._balance) {
        throw new InsufficientBalanceException("ยอดเงินไม่เพียงพอสำหรับการถอน");
      } else if (amount <= 0) {
        console.log("จำนวนเงินที่ถอนต้องมากกว่าศูนย์");
      } else {
        this._balance -= amount;
        console.log(`ถอนเงิน ${amount} บาท สำเร็จ. ยอดเงินปัจจุบัน: ${this._balance} บาท`);
      }
    }
  
    
    checkBalance() {
      console.log(`ยอดเงินในบัญชี ${this._holderName} (${this._accountNumber}): ${this._balance} บาท`);
    }
  }
  
  
  try {
    const account1 = new Account("123456789", "สมชาย", 1000.0);
  

    account1.checkBalance();
  

    account1.deposit(500.0);
  

    account1.withdraw(200.0);
  

    account1.checkBalance();
  

    account1.withdraw(2000.0);
  
  } catch (e) {
    if (e instanceof InsufficientBalanceException) {
      console.log(e.message);
    } else {
      console.error(e);
    }
  }
  

File: 051/4.py
class InsufficientBalanceException(Exception):
    pass

# Account class
class Account:
    def __init__(self, accountNumber, holderName, balance=0.0):
        self.__accountNumber = accountNumber
        self.__holderName = holderName
        self.__balance = balance

    # Getter and Setter for accountNumber
    def getAccountNumber(self):
        return self.__accountNumber

    def setAccountNumber(self, accountNumber):
        self.__accountNumber = accountNumber

    # Getter and Setter for holderName
    def getHolderName(self):
        return self.__holderName

    def setHolderName(self, holderName):
        self.__holderName = holderName

    # Getter and Setter for balance
    def getBalance(self):
        return self.__balance

    def setBalance(self, balance):
        self.__balance = balance

    # Method to deposit money
    def deposit(self, amount):
        if amount > 0:
            self.__balance += amount
            print(f"ฝากเงิน {amount} บาท สำเร็จ. ยอดเงินปัจจุบัน: {self.__balance} บาท")
        else:
            print("จำนวนเงินฝากต้องมากกว่าศูนย์")

    # Method to withdraw money
    def withdraw(self, amount):
        if amount > self.__balance:
            raise InsufficientBalanceException("ยอดเงินไม่เพียงพอสำหรับการถอน")
        elif amount <= 0:
            print("จำนวนเงินที่ถอนต้องมากกว่าศูนย์")
        else:
            self.__balance -= amount
            print(f"ถอนเงิน {amount} บาท สำเร็จ. ยอดเงินปัจจุบัน: {self.__balance} บาท")

    # Method to check balance
    def checkBalance(self):
        print(f"ยอดเงินในบัญชี {self.__holderName} ({self.__accountNumber}): {self.__balance} บาท")

# Demonstration of Bank Management System
try:
    # Create an account object
    account1 = Account("123456789", "สมชาย", 1000.0)

    # Check balance
    account1.checkBalance()

    # Deposit money
    account1.deposit(500.0)

    # Withdraw money
    account1.withdraw(200.0)

    # Check balance
    account1.checkBalance()

    # Try to withdraw more than available balance
    account1.withdraw(2000.0)

except InsufficientBalanceException as e:
    print(e)
